Utilities/build_ubuntu_cross_compilation_toolchain

```diff
@@ -0,0 +1,254 @@
+#!/bin/bash
+
+set -eu
+
+export PATH="/bin:/usr/bin"
+
+function usage() {
+    echo >&2 "Usage: $0 SWIFT-FOR-MACOS.pkg SWIFT-FOR-LINUX.tar.gz"
+    echo >&2
+    echo >&2 "Example: $0 /tmp/ ~/Downloads/swift-3.1-RELEASE-osx.pkg ~/Downloads/swift-3.1-RELEASE-ubuntu16.04.tar.gz"
+    echo >&2
+    echo >&2 "Complete example:"
+    echo >&2 "  # Download the Swift binaries for Ubuntu and macOS"
+    echo >&2 "  curl -o ~/Downloads/swift-3.1-RELEASE-ubuntu16.04.tar.gz https://swift.org/builds/swift-3.1-release/ubuntu1604/swift-3.1-RELEASE/swift-3.1-RELEASE-ubuntu16.04.tar.gz"
+    echo >&2 "  curl -o ~/Downloads/swift-3.1-RELEASE-osx.pkg https://swift.org/builds/swift-3.1-release/xcode/swift-3.1-RELEASE/swift-3.1-RELEASE-osx.pkg"
+    echo >&2 "  # Compile the SDK and toolchain from that"
+    echo >&2 "  $0 /tmp/ ~/Downloads/swift-3.1-RELEASE-osx.pkg ~/Downloads/swift-3.1-RELEASE-ubuntu16.04.tar.gz"
+    echo >&2 "  # Create a test application"
+    echo >&2 "  mkdir my-test-app"
+    echo >&2 "  cd my-test-app"
+    echo >&2 "  swift package init --type=executable"
+    echo >&2 "  # Build it for Ubuntu"
+    echo >&2 "  swift build --destination /tmp/cross-toolchain/ubuntu-xenial-destination.json"
+
+}
+
+if [[ $# -ne 3 ]]; then
+    usage
+    exit 1
+fi
+
+function realpath() {
+    if [[ "${1:0:1}" = / ]]; then
+        echo "$1"
+    else
+        (
+        cd "$(dirname "$1")"
+        echo "$(pwd)/$(basename "$1")"
+        )
+    fi
+}
+
+function fix_glibc_modulemap() {
```

Comment:
Is this something we expect to always have to do, or should there be a bug tacking eliminating the need for it? (i.e., is it something we fix in Swift)